mailpoet/lib/WooCommerce/Subscription.php

```diff
@@ -169,12 +169,14 @@ public function isCurrentUserSubscribed() {
     if (!$subscriber instanceof SubscriberEntity) {
       return false;
     }
+
     $wcSegment = $this->segmentsRepository->getWooCommerceSegment();
-    $subscriberSegment = SubscriberSegment::where('subscriber_id', $subscriber->getId())
-      ->where('segment_id', $wcSegment->getId())
-      ->findOne();
-    return $subscriberSegment instanceof SubscriberSegment
-      && $subscriberSegment->status === SubscriberEntity::STATUS_SUBSCRIBED;
+    $subscriberSegment = $this->subscriberSegmentRepository->findOneBy(
+      ['subscriber' => $subscriber->getId(), 'segment' => $wcSegment->getId()]
```

Comment:
I'm not sure, but maybe you don't know it. Passing IDs here is not necessary because the whole entity is enough, but you don't have to change.

Reply:
I wasn't aware of that. Thanks for mentioning it.